lib/mcb/commands/courses/create.rb

```diff
@@ -0,0 +1,184 @@
+      menu.default = :pgce_with_qts
+    end
+  end
+
+  def ask_study_mode
+    @study_mode = @cli.choose do |menu|
+      menu.prompt = "Full time or part time? (Full time if blank)  "
+      menu.choices(*%w[full_time part_time full_time_or_part_time]) { |cmd| cmd.to_sym }
+      menu.default = :full_time
+    end
+  end
+
+  def ask_age_range
+    @age_range = @cli.choose do |menu|
+      menu.prompt = "What's the level of the course?  "
+      menu.choices(:primary, :secondary, :middle_years, :other) { |cmd| cmd.to_sym }
+    end
+  end
+
+  def ask_accredited_body
+    code = @cli.ask("Provider code of accredited body (leave blank if self-accredited)  ")
+    @accredited_body = code.present? ? Provider.find_by(provider_code: code) : @provider
+  end
+
+  def ask_start_date
+    @start_date = Date.parse(@cli.ask("Start date?  ") {|q| q.default = "Sept 2019" })
+  end
+
+  def ask_course_code
+    @course_code = @cli.ask("Course code?  ")
+  end
+
+  def ask_ucas_subjects
+    finished = false
+    until finished
+      @cli.choose do |menu|
+        subject_list = !@ucas_subjects.empty? ? "(#{@ucas_subjects.map(&:subject_name).join(", ")} so far)" : ""
+        menu.prompt = "UCAS subjects to assign?#{subject_list}  "
+
+        subject_choices = @possible_ucas_subjects.order(:subject_name).pluck(:subject_name)
+        menu.choices(*(subject_choices + ['no more subjects'])) do |cmd|
+          cmd == 'no more subjects' ? finished = true : @ucas_subjects << @possible_ucas_subjects.detect {|s| s.subject_name == cmd }
+        end
+      end
+    end
+  end
+
+  def ask_sites
+    finished = false
+    until finished
+      @cli.choose do |menu|
+        sites = !@sites.empty? ? "(#{@sites.map(&:location_name).join(", ")} so far)" : ""
+        menu.prompt = "Which training locations to assign?#{sites}  "
+
+        site_choices = @possible_sites.order(:location_name).pluck(:location_name)
+        menu.choices(*(site_choices + ['no more training locations'])) do |cmd|
+          cmd == 'no more training locations' ? finished = true : @sites << @possible_sites.detect {|s| s.location_name == cmd }
+        end
+      end
+    end
+  end
+end
+
+class CoursePreview
+  def initialize(course)
+    @course = course
+  end
+
+  def to_h
+    {
+      "Age range" => @course.age_range,
+      "Course code" => @course.course_code,
+      "Start date" => @course.start_date.strftime("%b %Y"),
+      "Description" => @course.description,
+      "Provider" => @course.provider.provider_code,
+      "Accredited body" => @course&.accrediting_provider.provider_code || "Self-accrediting",
+      "Route" => @course.program_type,
+      "Modular" => @course.modular,
+      "English" => @course.english,
+      "Maths" => @course.maths,
+      "Science" => @course.science,
+      "Subjects" => @course.dfe_subjects.join(", "),
+      "Level" => @course.dfe_level,
+      "Training locations" => @course.sites.map(&:location_name).join(", "),
+    }
+  end
+end
```

Comment:
Feels like this is an opportunity to establish a pattern of reuse for the `show` commands we've already been adding.

Reply:
what did you have in mind?

Reply:
Just to reuse `CoursePreview` here and in `mcb courses show`, so maybe it needs to be `CourseShow` in `lib/mcb/course_show.rb` or something like that.

Reply:
Sure – I extracted the `CourseShow` as part of #326.